### PR TITLE
WIP Fix less (~...) with the mouse ##cons

### PR DESCRIPTION
**Pull request by 08A**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**
When using ~... with a cmd, if we use the mouse we get weird input. The error comme from line 1288 of the file libr/cons/dietline.c. We need to call r_cons_enable_mouse with false. The only problem
is the weird hack which interpret the struct I.Hud as a boolen (WTF ...).
We can't set this value to null. Indeed it is used to bufferize the input.
So i set the value to false but i need your help to understand why the line exist?

**Test plan**

aa
afl~... (click with your mouse)

**Closing issues**
#16790

### CODE REVIEW

File: libr/cons/hud.c [hunk 1]
```diff
@@ -45,7 +45,9 @@ R_API char *r_cons_hud_string(const char *s) {
 			os = o + i + 1;
 		}
 	}
+
```
Comment (trufae) [inline libr/cons/hud.c:48]: remove unnecessary empty lines

File: libr/cons/dietline.c [hunk 2]
```diff
@@ -1285,7 +1285,8 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 		__print_prompt ();
 	}
 	r_cons_break_push (NULL, NULL);
-	r_cons_enable_mouse (I.hud);
+	// r_cons_enable_mouse (I.hud);
+	r_cons_enable_mouse (false);
```
Comment (trufae) [inline libr/cons/dietline.c:1289]: I.hud is a struct, i so the commented line was related to the vi mode, which seems wrong to me. so I would probably say that its fine to remove the commented line.

**Comment by 08A** [comment]
> i didn't had a chance to test this yet. Please use `WIP` instead of --WIP-- and add ##cons in the PR title

What do you mean by '##cons'?

**Comment by trufae** [comment]
Ive tested your change and doesnt looks good to me..

- disabling the mouse makes the wheel scroll the terminal
- mouse wheel should scroll the list of filtered results like it does now

can you try this moving the set_raw() and mouse_enable(I.hud)

right now its buggy , so i can repro your issue:

* scroll down the list as expected
* scroll up scrolls the terminal

**Comment by 08A** [comment]
> Ive tested your change and doesnt looks good to me..
> 
>     * disabling the mouse makes the wheel scroll the terminal
> 
>     * mouse wheel should scroll the list of filtered results like it does now
> 
> 
> can you try this moving the set_raw() and mouse_enable(I.hud)

Can you be more precise?

> right now its buggy , so i can repro your issue:
> 
>     * scroll down the list as expected
> 
>     * scroll up scrolls the terminal

```
radare2 4.5.0-git 24723 @ linux-x86-64 git.4.4.0-243-gae2754a99
commit: ae2754a99ba178129d971a5e01e1ea40762c982d build: 2020-06-05__17:13:41
```
When i scroll up or down, there is just garbage value. (For me on master the issue is still there)

You `can't` reproduce the issue right?

**Comment by ret2libc** [comment]
@08A is this still WIP/draft? Can we review it?

**Comment by 08A** [comment]
@ret2libc It is WIP, i don't understand the code base. If someone want to take this issue, there is no problem with that.

**Comment by ret2libc** [comment]
@08A can I close this or do you plan to continue it?

**Comment by 08A** [comment]
I close this. I don't understand how to fix this hud.

**Pull request closed without merging**